src/IECoreImage/ImageReader.cpp

```diff
@@ -343,34 +326,44 @@ class ImageReader::Implementation
 		// Exception is thrown rather than false being returned.
 		bool open( bool throwOnFailure = false )
 		{
-			if( m_inputFile && m_reader->fileName() == m_inputFileName )
+			if( m_cache && m_reader->fileName() == m_inputFileName )
 			{
 				// we already opened the right file successfully
 				return true;
 			}
 
 			m_inputFileName = "";
-			m_inputFile.reset( ImageInput::open( m_reader->fileName() ) );
-			if( !m_inputFile )
+			m_cache.reset( ImageCache::create( /* shared */ false ) );
+
+			if( ImageInput *input = ImageInput::create( m_reader->fileName() ) )
 			{
-				if( !throwOnFailure )
-				{
-					return false;
-				}
-				else
-				{
-					throw IOException( string( "Failed to open file \"" ) + m_reader->fileName() + "\". " + geterror() );
-				}
+				m_cache->add_file( m_inputFileName );
```

Comment:
The docs say "There is no reason to use this method unless you are supplying a custom creator, or configuration, or both." Is there a reason other than that?

Reply:
Addressed in 73be01e